frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -138,12 +145,38 @@ export const useUserSettings = <T>(
   }, [cfLoaded, cfLoadError]);
 
   React.useEffect(() => {
-    if (cfData && cfLoaded && settings !== undefined) {
-      updateConfigMap(cfData, keyRef.current, seralizeData(settings));
+    if (sync && !isRequestPending.current) {
+      if (
+        cfData &&
+        cfLoaded &&
+        seralizeData(settingsRef.current) !== cfData?.data?.[keyRef.current]
+      ) {
+        setSettings(deseralizeData(cfData?.data?.[keyRef.current]));
+      }
     }
-    // This effect should only be run on change of settings state.
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [settings]);
+  }, [cfData, cfLoaded, sync]);
+
+  const callback = React.useCallback<React.Dispatch<React.SetStateAction<T>>>(
+    (action: React.SetStateAction<T>) => {
+      if (isRequestPending.current) return;
```

Comment:
Why does it matter if there's a request pending? what harm is there in saving the next value?